src/tunnelbridge_cmd.cpp

```diff
@@ -1574,11 +1554,36 @@ static BridgePieces CalcBridgePiece(uint north, uint south)
 	}
 }
 
+/**
+ * Get pillar information for a bridge middle tile.
+ * @param tile Tile of bridge middle.
+ * @param rampnorth Northern ramp tile of bridge.
+ * @param rampsouth Southern ramp tile of bridge.
+ * @param type Bridge type.
+ * @param transport_type Transport type of bridge.
+ * @return Pillar flags for bridge middle.
+ */
+static BridgePillarFlags GetBridgeTilePillarFlags(TileIndex tile, TileIndex rampnorth, TileIndex rampsouth, BridgeType type, TransportType transport_type)
+{
+	if (transport_type == TRANSPORT_WATER) return BRIDGEPILLARFLAGS_ALL_CORNERS;
+
+	const BridgeSpec *spec = GetBridgeSpec(type);
+	if (!spec->ctrl_flags.Test(BridgeSpecCtrlFlag::InvalidPillarFlags)) {
+		BridgePieces piece = CalcBridgePiece(GetTunnelBridgeLength(tile, rampnorth) + 1, GetTunnelBridgeLength(tile, rampsouth) + 1);
+		Axis axis = TileX(rampnorth) == TileX(rampsouth) ? AXIS_Y : AXIS_X;
+
+		return spec->pillar_flags[piece][axis == AXIS_Y ? 1 : 0];
```

Comment:
```suggestion
		return spec->pillar_flags[piece][TileX(rampnorth) == TileX(rampsouth)];
```

I'm not sure this actually improves things, but the intermediate Axis variable that's just used as a boolean flag feels a bit odd

Reply:
The current code makes it clearer that we are interested in the axis of the bridge. Your suggestion seems to hide that.